### PR TITLE
feat: message metada

### LINKED ISSUE #47: Rework message metadata
Provide type and API for sending and receiving message metadata. Think about "user metadata".

### PR DESCRIPTION
**Pull request by pchalupa**
GH card ticket "link": #ADD LINK TO GITHUB TICKET OR ISSUE

### Description

Resolves #47

### CODE REVIEW

File: apps/example/src/components/Chat.tsx [hunk 1]
```diff
@@ -1,13 +1,28 @@
-import { useEffect, useState } from 'react';
-import { useRtc, type EventHandler, type Message } from '@torolocos/react-rtc';
+import { useEffect, useState, useRef } from 'react';
+import { type RtcEvent, Message, useRtc } from '@torolocos/react-rtc';
 import './styles.css';
 
+type MessageMetadata = { username?: string };
+
+const isMessageMetadata = (
+  message: unknown
```
Comment (BrantalikP) [inline apps/example/src/components/Chat.tsx:8]: Is this Guard really necessary?  :sweat_smile:

Comment (pchalupa) [reply apps/example/src/components/Chat.tsx:8]: Nice catch! The short answer is yes. There is why I think it:

We don't know the shape of metadata = type is `unknown` https://github.com/torolocos/react-rtc/blob/6e850dfd9e5e1cc1ee11d75102a3ca279b226709/packages/react-rtc/src/models/Message.ts

We let people define their shape/data so we should also "force" them to do runtime checks.

File: apps/example/src/components/Chat.tsx [hunk 2]
```diff
@@ -23,28 +36,38 @@ const Chat = () => {
...
+    if (isMessage(event.detail)) {
+      const message = event.detail;
 
-  const handleMessageSend: EventHandler<'send'> = (event) =>
-    setMessageData((messages) => [...messages, event.detail]);
+      setMessageData((messages) => [...messages, message]);
+    }
+  };
+
+  const handleMessageSend = (event: RtcEvent<'send'>) => {
```
Comment (BrantalikP) [inline apps/example/src/components/Chat.tsx:52]:
```suggestion
  const handleMessageSent= (event: RtcEvent<'send'>) => {
```

Comment (BrantalikP) [reply apps/example/src/components/Chat.tsx:52]: not sure about this, but if there is past tense `handleMessageReceived `

**Review by BrantalikP — APPROVED** [review]
:clap:

**Review by BrantalikP — APPROVED** [review]

**Pull request merged**